components/function-controller/pkg/apis/serverless/v1alpha1/gitrepository_validation.go

```diff
@@ -33,3 +38,27 @@ func (in *RepositoryAuth) validate(path string) *apis.FieldError {
 		value: string(in.Type),
 	})
 }
+
+func (in *RepositoryAuth) validateSSHAuth(path string) *apis.FieldError {
+	if in == nil {
+		return apis.ErrMissingField(path)
+	}
+	var err *apis.FieldError
+	if in.Type != RepositoryAuthSSHKey {
+		err = apis.ErrGeneric(fmt.Sprintf("invalid value for git ssh, expected %s, current: %s",
+			RepositoryAuthSSHKey, in.Type), fmt.Sprintf("%s.type", path))
+	}
+
+	return err.Also(validateIfMissingFields(property{
+		name:  fmt.Sprintf("%s.secretName", path),
+		value: in.SecretName,
+	}))
+}
+func isRepoURLIsSSH(repoURL string) bool {
```

Comment:
Could you please write some more tests for that regexp? In first glance it looks kinda complicated

Reply:
Done